docs/reference/sql/functions/date-time.asciidoc

```diff
@@ -404,6 +404,48 @@ include-tagged::{sql-specs}/docs/docs.csv-spec[dateDiffDateTimeMinutes]
 include-tagged::{sql-specs}/docs/docs.csv-spec[dateDiffDateMinutes]
 --------------------------------------------------
 
+[[sql-functions-datetime-dateformat]]
+==== `DATE_FORMAT/FORMAT/TO_CHAR/TOCHAR`
```

Comment:
Do all the function names exist in other DBs? I'd like one which is unique to us due to the Java time format - maybe `datetime_format` (and `datetime_parse`) - or vice-versa - `format_datetime` and `parse_datetime`.

My point is, I think going forward it would be good to have our own function which uses Java time and then support the rest of the DB functions as translating implementation for the given arguments to our class.
For example `DATE_FORMAT` which is MySQL specific would exist as such but the MySQL arguments would then be converted into the `java.time` pattern.
Otherwise looking like MySQL but not supporting its pattern is confusing.

Since the pattern support is tedious, I'm fine with doing that in a separate support for the future. However I'd like to clarify the approach in the docs.

MySQL `DATE_FORMAT`: https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format
SQL Server `FORMAT`: https://docs.microsoft.com/en-us/sql/t-sql/functions/format-transact-sql?view=sql-server-ver15#ExampleD
Postgres `TO_CHAR`: https://www.postgresql.org/docs/9.1/functions-formatting.html
Oracle `TOCHAR`: https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements004.htm#i34510

Note that outside MySQL, the rest of the functions accept multiple formats and data types so to avoid confusion, I would not create aliases for them yet.
But rather treat them separately since if we are to introduce such functions, we would essentially have to obey their semantics rather then ours.

Reply:
All of `DATE_FORMAT`, `FORMAT` and `TO_CHAR` and `TOCHAR` are used in DBs and I agree with the confusion regarding the patterns so I'd go for `datetime_format/parse` with our own ES-SQL impl (java.time).

@astefan @bpintea Do you agree with the naming?

Reply:
Renamed to `DATETIME_FORMAT`

Reply:
Changed now, but it seems sane to me, 👍.